source/libraries/ContractDeployer.ts

```diff
@@ -0,0 +1,231 @@
+        const constructorSignature = <AbiFunction | undefined>abi.find(signature => signature.type === 'constructor');
+        if (typeof constructorSignature === 'undefined') throw new Error(`ABI did not contain a constructor.`);
+        const constructorInputTypes = constructorSignature.inputs.map(x => x.type);
+        const encodedConstructorParameters = Buffer.from(encodeParams(constructorInputTypes, constructorArgs).substring(2), 'hex');
+        return Buffer.concat([bytecode, encodedConstructorParameters]);
+    }
+
+    private async construct(contract: Contract, constructorArgs: Array<string>, failureDetails: string): Promise<string> {
+        const data = `0x${ContractDeployer.getEncodedConstructData(contract.abi, contract.bytecode, constructorArgs).toString('hex')}`;
+        // TODO: remove `gas` property once https://github.com/ethereumjs/testrpc/issues/411 is fixed
+        const gasEstimate = await this.connector.ethjsQuery.estimateGas({
+            from: this.accountManager.defaultAddress,
+            data: data
+        });
+        const nonce = await this.accountManager.nonces.get(this.accountManager.defaultAddress);
+        const signedTransaction = await this.accountManager.signTransaction({
+            gas: gasEstimate,
+            gasPrice: this.connector.gasPrice,
+            data: data
+        });
+        console.log(`Upload contract: ${contract.contractName} nonce: ${nonce}, gas: ${gasEstimate}, gasPrice: ${this.connector.gasPrice}`);
+        const transactionHash = await this.connector.ethjsQuery.sendRawTransaction(signedTransaction);
+        const receipt = await this.connector.waitForTransactionReceipt(transactionHash, failureDetails);
+        console.log(`Uploaded contract: ${contract.contractName}: \"${receipt.contractAddress}\"`);
+        return receipt.contractAddress;
+    }
+
+    private async generateAddressMapping(contractAddressMapping: ContractAddressMapping): Promise<string> {
+        const networkId = await this.connector.ethjsQuery.net_version();
+        let addressMapping: NetworkAddressMapping = {};
+        if (await exists(this.configuration.contractAddressesOutputPath)) {
+            let existingAddressFileData: string = await readFile(this.configuration.contractAddressesOutputPath, 'utf8');
+            addressMapping = JSON.parse(existingAddressFileData);
+        }
+        addressMapping[networkId] = contractAddressMapping;
+        return JSON.stringify(addressMapping, null, ' ');
+    }
+
+    private async generateAddressMappingFile(contractAddressMapping: ContractAddressMapping): Promise<void> {
+        const addressMappingJson = await this.generateAddressMapping(contractAddressMapping);
+        await writeFile(this.configuration.contractAddressesOutputPath, addressMappingJson, 'utf8')
+    }
+
+//
+//     private async generateUploadBlockNumberMapping(blockNumber: number): Promise<string> {
```

Comment:
Did you mean to leave this commented out?  Can it be deleted instead?  Perhaps something you mean to get back to after this PR?

Reply:
Are we ever going to getLogs? If not, we don't need to worry about upload block numbers.

Reply:
I can't think of any reason we would want to, and even if we did for now I think it is probably easier to just manually deal with it rather than automatically deal with it.